EKF/control.cpp

```diff
@@ -1356,6 +1356,13 @@ void Ekf::controlMagFusion()
 	// check for new magnetometer data that has fallen behind the fusion time horizon
 	// If we are using external vision data for heading then no magnetometer fusion is used
 	if (!_control_status.flags.ev_yaw && _mag_data_ready) {
+		// perform a yaw reset if requested by other functions
+		if (_mag_yaw_reset_req) {
+			if (!_mag_use_inhibit ) {
```

Comment:
I'm just curious, why is _mag_use_inhibit only checked in this reset case? There are quite a few other calls to reset (as I'm sure you know).

Reply:
That variable was introduced as part of the protection against indoor magnetic field errors and can only be true if _params.mag_fusion_type == MAG_FUSE_TYPE_INDOOR.

_mag_use_inhibit  is in used inside fuseHeading() to prevent a magnetic heading being used and causing either a static heading of 0 or a zero innovation to be used during heading fusion.

In theory a mag heading reset should not be called if _mag_use_inhibit  is true, however by introducing that constraint to the other resets that are not part of this fix we could be unintentionally introducing a bug due to a possible race condition when GPS or EV use is gained/lost. I wanted to keep the scope of this change minimal to reduce possibility of unintended consequences.